src/worker_pool/index.ts

```diff
@@ -31,6 +33,8 @@ export class WorkerInfo extends AsynchronouslyCreatedResource {
     lastSeenResponseCount : number = 0;
     onMessage : ResponseCallback;
     histogram: RecordableHistogram | null;
+    terminating = false;
+    destroyed = false;
```

Comment:
What are the use cases needing to keep track of the worker lifecycle at termination or pool destroying? 

The use cases regarding worker state signaling: 

- Is a worker ready?
- Is a worker errored?

Reply:
The use-cases I've in mind was mostly monitoring or possibly to update the load balancer metrics upon request, my fear here is to cause big overhead due to the constant creations of EventEmitter.

Although the API might be better to communicate this, I found enough by stating the current state of the worker, found it less intrusive and easier to keep track. As using `getters` for it, the state will be updated almost immediately, reflecting the state of the worker at a given point of time.

Maybe I need to add some more to it, but currently I could just think on terminating and destroyed; which can be potentially helpful as there will be race conditions where the balancer might want to distribute the workload to a given worker that its crashing at the same time (reducing the surface but maybe not mitigating it).

Reply:
Understood now. Make sense. The implementation is debatable but is inline with the existing events emission one.

LB would need to track worker lifecycle and task(s) lifecycle on a worker.